### PR TITLE
read unsigned int in deh_GetData

### PR DESCRIPTION
**Pull request by rfomin**
Fixes this issue: [[doomworld]](https://www.doomworld.com/forum/topic/123082-heretic-invades-doom-two-small-maps-for-mbf21-ports/?tab=comments#comment-2345308)

### CODE REVIEW

File: Source/d_deh.c [hunk 1]
```diff
@@ -3043,7 +3044,7 @@ char *ptr_lstrip(char *p)  // point past leading whitespace
 boolean deh_GetData(char *s, char *k, long *l, char **strval, FILE *fpout)
 {
   char *t;  // current char
-  long val; // to hold value of pair
+  int val; // to hold value of pair
```
Comment (fabiangreffrath) [inline Source/d_deh.c:3047]: Hm, why not `unsigned int`?

Comment (rfomin) [reply Source/d_deh.c:3047]: This is what I want to ask: Should we change all flags related ints to uint32_t?

Comment (fabiangreffrath) [reply Source/d_deh.c:3047]: Type `unsigned int` should be fine (at least it is not as ambigious as `long`), not sure if we have C99-style `uint32_t` types elsewhere.

Comment (fabiangreffrath) [reply Source/d_deh.c:3047]: PS: You might need to cast `(int *) &val` in the `M_StrToInt()` call then.

Comment (rfomin) [reply Source/d_deh.c:3047]: > not sure if we have C99-style `uint32_t` types elsewhere

We have. We are also using Uint32 from SDL.

Comment (fabiangreffrath) [reply Source/d_deh.c:3047]: Alright, whatever makes more sense to you then. 😉

Comment (rfomin) [reply Source/d_deh.c:3047]: Honestly, I don't want to change anything :smile: Could we have problems because of this?

Comment (fabiangreffrath) [reply Source/d_deh.c:3047]: Not sure, type conversions in C are tricky. But if you say it solves the bug reported in this thread, let's take it as it is.

On 32 bit systems and on Windows, `long` and `int` are identical anyway, I'll have to check at home what happens on 64 bit Linux.

**Pull request merged**